README


This archive contains the distribution Courriel,
version 0.35:

  High level email parsing and manipulation

This software is Copyright (c) 2014 by Dave Rolsky.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)


This README file was generated by Dist::Zilla::Plugin::Readme v5.016.



Changes
0.36     2014-08-15

- Trying to build an email with a header that started with a space caused the
  header value to be dropped entirely from the generated email. Patch by Greg
  Oschwald. PR #2.


0.35     2014-05-12

- Replaced all uses of File::Slurp with File::Slurp::Tiny. See RT #95479 for
  the initial bug report and discussion on why File::Slurp should not be used.


0.34     2014-05-02

- More test fixes - relax the date parsing more.


0.33     2014-04-19

- Fixed tests so they pass with the most recent DateTime::Format::Mail (0.400)
  as well as older versions.


0.32     2014-04-19

- Courriel parsing is now more liberal and will accept an email with mixed
  line endings (some newline, some CRLF, etc.). Apparently this happens in the
  wild (of course it does). Reported by Joseph Crotty. RT #94837.


0.31     2013-04-03

[MISC]

- Removed the "this is very alpha" warning from the docs. It's really not very
  alpha any more.


0.30     2012-08-20

[API CHANGES]

- Courriel::Builder no longer exports a bcc() subroutine. Creating Bcc headers
  in emails does not make any sense whatsoever. See
  https://metacpan.org/module/Email::Sender::Manual::QuickStart#Hey-wheres-my-Bcc-support-
  for an explanation of why there is no such thing as a Bcc header.


0.29     2012-03-07

[BUG FIXES]

- Courriel now allows an email to have an empty subject when parsing, as does
  Courriel::Builder. Based on a patch from Zbigniew Łukasiak.


0.28     2012-02-01

[BUG FIXES]

- Header attribute names need to be looked up case-insensitively, according to
  the spec. The original casing is preserved in the Courriel::HeaderAttribute
  object's name(), method, but looking up an attribute by name on a
  Courriel::Header object now accepts any casing for the name.


0.27     2011-09-24

[ENHANCEMENTS]

- All objects which can stringify now support a ->stream_to API as well. This
  lets you stream their string output to a sub ref, filehandle, or an object
  with a ->print() method. If you are dealing with large emails, this API may
  be more efficient than generating a single string in memory.

[API CHANGES]

- Renamed the Header->as_header_string and HeaderAttribute->as_header_string
  methods to as_string, so that all objects have the same stringification API.


0.26     2011-09-19

[BUG FIXES]

- Add missing prereq for MooseX::Role::Parameterized. Reported by zby.


0.25     2011-09-15

[BUG FIXES]

- When a header value contained any non-ASCII content, Courriel would
  mistakenly MIME encode the whole value. In some cases, this is okay, but for
  some header fields it is wrong. For example, when a field contains an email
  address the actual address portion must be unencoded.


0.24     2011-09-13

[ENHANCEMENTS]

- The is_binary flag has been replaced with an is_character flag. This flag
  defaults to false. Suggested by zby.


0.23     2011-09-13

[ENHANCEMENTS]

- Courriel->parse now accepts an is_binary flag so you can pass decoded
  (utf-8) data as the email text. However, it's probably smarter to simply
  pass binary data to Courriel and let it do the decoding. The issue was
  brought up by zby.


0.22     2011-09-13

[BUG FIXES]

- Require List::MoreUtils 0.28+, as tests fail with earlier versions.

- Added missing prereq on File::Slurp.


0.21     2011-09-12

[API CHANGES]

- The ->get_attribute() method for Courriel::ContentType and ::Disposition
  objects was replaced by ->attribute in 0.20.

[ENHANCEMENTS]

- Added a new ->get_values() method to Courriel::Headers. This lets you get
  the header string values directly.

- Added a new ->attribute_value() for Courriel::Header::ContentType and
  ::Disposition. This lets you get the attribute's string value, if the
  attribute exists. Suggested by zby.

[BUG FIXES]

- The previous release documented a method ->get_attribute() for
  Courriel::Header::ContentType and ::Disposition objects. This method was
  removed in 0.20 and replaced with a method named ->attribute().


0.20     2011-09-11

[API CHANGES]

- The way headers are represented has been changed quite a bit
  internally. Read on for details.

  All headers are now represented as Courriel::Header objects instead of plain
  strings. This means that calling $headers->get('Foo') returns objects. To
  get the value, you need to call ->value on the individual objects.

  See Courriel::Header for the new class for headers.

- As a corrolary, the Courriel::ContentType and Courriel::Disposition classes
  have been renamed to Courriel::Header::ContentType and
  Courriel::Header::Disposition. They are both now subclasses of
  Courriel::Header. The APIs of these classes have remained mostly the same,
  except for attribute handling (see below).

  The old classes are still shipped in the distro, but they die when
  loaded. This avoids leaving behind old modules which no longer work with
  newer versions of Courriel.

  Methods which returned the old object now return the new one. The
  Courriel::Headers class will return the appropriate object when the header
  is fetched with $headers->get().

- Header attributes for the Content-Type and Content-Disposition object are
  now represented as Courriel::HeaderAttribute objects. This is necessary
  because these attributes may contain information about their character set
  and language, per RFC2231.

[ENHANCEMENTS]

- Header attributes with character set and language data per RFC2231 are now
  parsed properly. Reported by zby. RT #70801.

- Header attribute parsing is much laxer and will not die on bad attributes
  syntax (in most cases). It simply stops parsing and ignores bad data.


0.19     2011-08-21

- Fixed some test failures with older versions of Test::More, which stringify
  objects passed to is() rather than using their overloading. Reported by
  Maximilian Gaß. RT #70395.

- The Courriel::Disposition methods which return DateTime objects now
  explicitly set the time zone of the objects to UTC before returning it.


0.18     2011-08-20


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "High level email parsing and manipulation",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Courriel",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Courriel",
  "PREREQ_PM" => {
    "Carp" => 0,
    "DateTime" => 0,
    "DateTime::Format::Mail" => 0,
    "DateTime::Format::Natural" => 0,
    "Devel::PartialDump" => 0,
    "Email::Abstract::Plugin" => 0,
    "Email::Address" => 0,
    "Email::MIME::Encodings" => 0,
    "Email::MessageID" => 0,
    "Encode" => 0,
    "Exporter" => 0,
    "File::Basename" => 0,
    "File::LibMagic" => 0,
    "File::Slurp::Tiny" => 0,
    "List::AllUtils" => 0,
    "List::MoreUtils" => "0.28",
    "MIME::Base64" => 0,
    "MIME::QuotedPrint" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "MooseX::Params::Validate" => 0,
    "MooseX::Role::Parameterized" => 0,
    "MooseX::StrictConstructor" => 0,
    "MooseX::Types" => 0,
    "MooseX::Types::Combine" => 0,
    "MooseX::Types::Common::String" => 0,
    "MooseX::Types::Moose" => 0,
    "Scalar::Util" => 0,
    "Sub::Exporter" => 0,
    "namespace::autoclean" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::Differences" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "Test::Requires" => 0,
    "utf8" => 0
  },
  "VERSION" => "0.36",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "DateTime" => 0,
  "DateTime::Format::Mail" => 0,
  "DateTime::Format::Natural" => 0,
  "Devel::PartialDump" => 0,
  "Email::Abstract::Plugin" => 0,
  "Email::Address" => 0,
  "Email::MIME::Encodings" => 0,
  "Email::MessageID" => 0,
  "Encode" => 0,
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Basename" => 0,
  "File::LibMagic" => 0,
  "File::Slurp::Tiny" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::AllUtils" => 0,
  "List::MoreUtils" => "0.28",
  "MIME::Base64" => 0,
  "MIME::QuotedPrint" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "MooseX::Params::Validate" => 0,
  "MooseX::Role::Parameterized" => 0,
  "MooseX::StrictConstructor" => 0,
  "MooseX::Types" => 0,
  "MooseX::Types::Combine" => 0,
  "MooseX::Types::Common::String" => 0,
  "MooseX::Types::Moose" => 0,
  "Scalar::Util" => 0,
  "Sub::Exporter" => 0,
  "Test::Differences" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.88",
  "Test::Requires" => 0,
  "namespace::autoclean" => 0,
  "parent" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



